pkg/machine/machineset.go

```diff
@@ -0,0 +1,581 @@
+	}
+
+	if builder.Definition == nil {
+		glog.V(100).Infof("The MachineSet object definition is nil")
+
+		builder.errorMsg = "MachineSet 'Object.Definition' is nil"
+	}
+
+	return &builder
+}
+
+// PullSet loads an existing MachineSet into Builder struct.
+func PullSet(apiClient *clients.Settings, name, namespace string) (*SetBuilder, error) {
+	glog.V(100).Infof("Pulling existing machineSet name %s in namespace %s", name, namespace)
+
+	builder := SetBuilder{
+		apiClient: apiClient,
+		Definition: &machinev1beta1.MachineSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+		},
+	}
+
+	if name == "" {
+		builder.errorMsg = "MachineSet 'name' cannot be empty"
+	}
+
+	if namespace == "" {
+		builder.errorMsg = "MachineSet 'namespace' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("machineSet object %s doesn't exist in namespace %s", name, namespace)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Exists checks whether the given MachineSet exists.
+func (builder *SetBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof(
+		"Checking if MachineSet %s exists in namespace %s", builder.Definition.Name,
+		builder.Definition.Namespace)
+
+	var err error
+	builder.Object, err = builder.apiClient.MachineSets(builder.Definition.Namespace).Get(context.TODO(),
+		builder.Definition.Name, metav1.GetOptions{})
+
+	if err != nil {
+		glog.V(100).Infof("Failed to collect MachineSet object due to %s", err.Error())
+	}
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Create makes a MachineSet in cluster and stores the created object in struct.
+func (builder *SetBuilder) Create() (*SetBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating the MachineSet %s", builder.Definition.Name)
+
+	var err error
+	if !builder.Exists() {
+		builder.Object, err = builder.apiClient.MachineSets(builder.Definition.Namespace).Create(
+			context.TODO(), builder.Definition, metav1.CreateOptions{})
+	}
+
+	return builder, err
+}
+
+// Delete removes a MachineSet object from a cluster.
+func (builder *SetBuilder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Deleting the MachineSet object %s",
+		builder.Definition.Name)
+
+	if !builder.Exists() {
+		return fmt.Errorf("machineSet cannot be deleted because it does not exist")
+	}
+
+	err := builder.apiClient.MachineSets(builder.Object.Namespace).Delete(
+		context.TODO(), builder.Object.Name, metav1.DeleteOptions{})
+
+	if err != nil {
+		return fmt.Errorf("cannot delete MachineSet: %w", err)
+	}
+
+	return err
+}
+
+// WaitForMachineSetReady waits until MachineSet first replica is Ready.
+func WaitForMachineSetReady(
+	apiClient *clients.Settings,
+	namespace,
+	machineSetName string,
+	replicas int32,
```

Comment:
Is replicas necessary here? Can't you just use machineSetPulled.Object.Status.Replicas? This is an example of how we do this in the deployment pkg - https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/deployment/deployment.go#L445